### ISSUE TITLE
Unhandled `async` while parsing items inside trait

### ISSUE THREAD

**Issue by braw-lee** [Rust-GCC/gccrs#2778]
[error](https://godbolt.org/z/5Kor8j5M9)
compiler does not recognize `async` as a keyword which is bug

**Comment by braw-lee** [comment]
I want to fix #2767 which will build upon this commit, should I wait for this PR to be merged and create a new PR for that issue
or do you want me to handle that issue in this PR?
@P-E-P

**Comment by P-E-P** [comment]
> I want to fix #2767 which will build upon this commit, should I wait for this PR to be merged and create a new PR for that issue
> or do you want me to handle that issue in this PR?
> @P-E-P 

We try to keep PRs short to make them easy to review. Usually when some changes depend on a given open PR people write something such as `depends on #XXXX` and keep the second PR in draft until the first one is accepted. They rebase the second PR on master once the first one is merged. We won't reject a PR because it solves two issues though.

**Comment by braw-lee** [comment]
ok thanks, 
since the commits were related and really small I add commits to same PR
